### PR TITLE
Moved MDAnalys related parameters into main

### LINKED ISSUE #36: Do not hardcode analysis modules
Currently the `relevant_modules` are hardcoded as global variable in `cli.py`. This is not only bad coding practice but also prevents third party libraries to use the cli for their Analysis modules. The lines

https://github.com/MDAnalysis/mdacli/blob/78fa3f2fff31fb7f9acf71ee8811ab902369e947/src/mdacli/cli.py#L29-L34

should be combined with 

https://github.com/MDAnalysis/mdacli/blob/78fa3f2fff31fb7f9acf71ee8811ab902369e947/src/mdacli/cli.py#L337

and moved into the main function. `setup_cli` should then take the list of modules as required argument.

### PR DESCRIPTION
**Pull request by PicoCentauri**
Fixes #36.

This PR moves all parameters directly related to MDAnalysis inside the main function and generalizes the setup_clients function. With this change it is much easier for downstream projects (like taurenmd and maicos) to use the parser for their Analysis classes.

### CODE REVIEW

File: src/mdacli/cli.py [hunk 1]
```diff
@@ -372,4 +366,24 @@ def setup_clients():
 
 def main():
     """Execute main CLI entry point."""
-    maincli(setup_clients())
+    # modules in MDAnalysis.analysis packages that are ignored by mdacli
+    # relevant modules used in this CLI factory
+    # hydro* are removed here because they have a different folder/file
+    # structure and need to be investigated separately
+    skip_mods = ('base', 'hydrogenbonds', 'hbonds')
+    relevant_modules = (_mod for _mod in __all__ if _mod not in skip_mods)
```
Comment (joaomcteixeira) [inline src/mdacli/cli.py:374]: I suggest keeping skip_mods and _relevant_modules as global values. I have already import them for some personal tests. I prefer them being global.

File: src/mdacli/cli.py [hunk 2]
```diff
@@ -372,4 +366,24 @@ def setup_clients():
...
+
+    members = []
+    for module in relevant_modules:
```
Comment (joaomcteixeira) [inline src/mdacli/cli.py:377]: Let's make this an individual function

**Review by joaomcteixeira — CHANGES_REQUESTED** [review]
Minor details.

File: src/mdacli/cli.py (hunk 2 above)
Comment (PicoCentauri) [reply src/mdacli/cli.py]: Done.

File: src/mdacli/cli.py (hunk 1 above)
Comment (PicoCentauri) [reply src/mdacli/cli.py]: I revert the change. If we want to keep this global we should also capitalize this right? BTW what about the dictionary. 

https://github.com/MDAnalysis/mdacli/blob/d89ae1b9763fd3968ca47bbea13e98448ee8c507/src/mdacli/cli.py#L39-L40

Do you also want to keep this global?

File: src/mdacli/libcli.py [hunk 3]
```diff
@@ -32,3 +44,30 @@ def __call__(self, parser, namespace, value, option_string=None):
                 ) from None
 
         setattr(namespace, self.dest, jdict)
+
+
+def find_AnalysisBase_members(*module_name):
```
Comment (joaomcteixeira) [inline src/mdacli/libcli.py:49]: because it is a list make it `module_names`. :picky me: :P

File: src/mdacli/cli.py (hunk 1 above)
Comment (joaomcteixeira) [reply src/mdacli/cli.py]: i am now on this

Comment (joaomcteixeira) [reply src/mdacli/cli.py]: refactored

File: src/mdacli/libcli.py [hunk 4]
```diff
@@ -32,3 +45,51 @@ def __call__(self, parser, namespace, value, option_string=None):
...
+    """
+    members = []
+    for name in module_names:
+        module = importlib.import_module(name)
+        for _, member in inspect.getmembers(module):
+            if inspect.isclass(member) and issubclass(member, klass) \
+               and member is not klass:
+                members.append(member)
+
+    return members or None
```
Comment (joaomcteixeira) [inline src/mdacli/libcli.py:78]: @PicoCentauri :eye: here

**Comment by joaomcteixeira** [comment]
@PicoCentauri 
very nice Pull Request.
Made some additional refactors for good and to facilitate the testing.
This PR is :+1: for me, merge it if you agree.

File: tests/test_cli.py [hunk 5]
```diff
@@ -56,8 +60,10 @@ def test_setup_clients(opt, dest, val):
             testargs.append(str(i))
     else:
         testargs.append(str(val))
+    members = find_AnalysisBase_members_ignore_warnings(_relev_mod)
```
[the remark this answers is not in the document]
Comment (PicoCentauri) [reply tests/test_cli.py:63]: I see the issue. Lets leave it open here and you can open an issue for this?

File: src/mdacli/utils.py [hunk 6]
```diff
@@ -152,12 +149,12 @@ def parse_callable_signature(callable_obj, storage_dict):
...
-    storage_dict[callable_obj.__name__]["optional"] = optional_args
-    storage_dict[callable_obj.__name__]["desc"] = summary
-    storage_dict[callable_obj.__name__]["desc_long"] = summary_extended
+    storage_dict["positional"] = positional_args
+    storage_dict["optional"] = optional_args
+    storage_dict["desc"] = summary
+    storage_dict["desc_long"] = summary_extended
 
-    return
+    return storage_dict
```
Comment (PicoCentauri) [inline src/mdacli/utils.py:157]: This looks much cleaner!

**Pull request merged**